File: BAEKJOON/java/10865_친구_친구.java
import java.io.*;
import java.util.StringTokenizer;

public class Main {
    public static int n, m;
    public static int[] friends = new int[100001];

    public static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        input();
        solution();

        bw.flush();
        bw.close();
    }

    private static void input() throws IOException {
        StringTokenizer st = new StringTokenizer(br.readLine());

        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());

        for (int i = 1; i <= m; i++) {
            st = new StringTokenizer(br.readLine());

            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());

            friends[a]++;
            friends[b]++;
        }
    }

    private static void solution() throws IOException {
       for (int i = 1; i <= n; i++) {
           bw.write(friends[i] + "\n");
       }
    }
}

File: BAEKJOON/cpp/10865_친구_친구.cpp
#include <iostream>
using namespace std;

int n, m;
int friends[100001];

void input() {
	cin >> n >> m;

	int a, b;

	for (int i = 1; i <= m; i++) {
		cin >> a >> b;
		friends[a]++;
		friends[b]++;
	}
}

void solution() {
	for (int i = 1; i <= n; i++) {
		cout << friends[i] << "\n";
	}
}

int main() {
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	input();
	solution();

	return 0;
}